app/assets/javascripts/master_detail.js

```diff
@@ -90,27 +90,31 @@ function clickBack(btn, level) {
 }
 
 function reset_grid(selector, level){
-  var $grid = $(selector);
-  var dataGrid = $grid.dxDataGrid('instance');
-  try {
-    if($grid.data("compact-view") && ($grid.data("compact-view")[level] || []).length > 0) {
-      $.each($grid.data("compact-view")[level], function(i, item) {
-        $grid.dxDataGrid('columnOption', item.name, item.property, item.value);
-      });
-    }
-  }
-  catch(e) {
-    // Catches the instance where the compact view is not defined in a master detail (but its indeterminable)
-  }
-  finally {
-    window.setTimeout(function(){
-      // If someone is clicking fast on different rows, the grid may not be on the dom anymore.
-      // We don't want to repaint the grid if it isn't there. Plus, it will fail if it isn't there.
-      if ($(selector).length > 0) {
-        dataGrid.updateDimensions();
-        dataGrid.repaint();
+  var $grid = $(selector);  
+  var dataGrid;
+
+  if($grid.isConnected) {
```

Comment:
I could be wrong, but...
From what I know, `isConnected` is a part of javascript Web Api, not part of jQuery. So, this is not going to work as you may expect.
I believe what is happening now, is that `$grid.isConnected` is returning `undefined` which is falsey. Which in turn means that it will never go into that block.

Reply:
You could try `document.QuerySelector(selector).isConnected`, but that should be redundant, as `document.QuerySelector(selector)` should only return the Node if it is present in the DOM.

I believe isConnected is there to check if a Node is attached to the dom through document. If you are using a selector to get the node/element/collection, and it returns it, then it should be attached to the dom.

Reply:
In short...
Maybe just add a guard clause that checks if `$grid.length < 1`.
And then remove the check for length in the `finally` block.